src/Package/Impl/Sql/Commands/PublishSProcCommand.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using Microsoft.Common.Core;
+using Microsoft.Common.Core.Shell;
+using Microsoft.Languages.Core.Settings;
+using Microsoft.VisualStudio.R.Package.Commands;
+using Microsoft.VisualStudio.R.Package.ProjectSystem;
+using Microsoft.VisualStudio.R.Package.Sql.Publish;
+using Microsoft.VisualStudio.Shell.Interop;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Utilities;
+using Microsoft.Languages.Editor.Composition;
+using Microsoft.R.Components.ContentTypes;
+#if VS14
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.ProjectSystem.Designers;
+#else
+using Microsoft.VisualStudio.ProjectSystem;
+#endif
+
+namespace Microsoft.VisualStudio.R.Package.Sql {
+    [ExportCommandGroup("AD87578C-B324-44DC-A12A-B01A6ED5C6E3")]
+    [AppliesTo(ProjectConstants.RtvsProjectCapability)]
+    internal sealed class PublishSProcCommand : ICommandGroupHandler {
+        private readonly ICoreShell _coreShell;
+        private readonly IProjectSystemServices _pss;
+        private readonly IWritableSettingsStorage _settingsStorage;
+
+        [ImportingConstructor]
+        public PublishSProcCommand(ICoreShell coreShell, IProjectSystemServices pss, [Import(AllowDefault = true)]IWritableSettingsStorage settingsStorage) {
+            if (settingsStorage == null) {
```

Comment:
I thought about that... Technically this is language services storage but I guess it is OK to use in general. No reason to create yet another one. I'll make it a member on core shell.